### ISSUE TITLE
Rationalise pseudopotential types

### ISSUE THREAD

**Issue by davidbowler** [OrderN/CONQUEST-release#202]
At present we have a redundant pseudopotential type (OLDPS) which I think that we should remove.  It would also be possible to change the code so that the `pseudo_type` variable becomes species-dependent, to allow mixing of Conquest and Siesta ion files.

@tsuyoshi38 @lionelalexandre @ayakon Do you have any opinions, particularly on allowing mixing of types?

**Comment by davidbowler** [comment]
Looking more carefully at this, it's clear that we can mix pseudopotential types when we use the neutral atom potential.  However it's harder if we don't: the calculation for the local part of the potential might cause problems with mixed types.  Do we want to allow mixed types only with neutral atom potential?

**Comment by tsuyoshi38** [comment]
I agree that removing OLDPS is okay.  (As far as I remember, it was easy to make compatible with other pseudo potentials like the ones used for STATE with OLDPS. But, we don't have PAOs for this.)

For mixing Siesta and Conquest ion files, 
since the expression of local potential is different, it is difficult to mix them without neutral atom (NA) potentials, but should be okay with NA.  (as you said. Sorry, I have not noticed your comment above..  > Dave.)
I can imagine that it would be useful in some cases if we can mix them, and 
it should be okay to assume users need to use NA potential in this case, I think.

**Comment by davidbowler** [comment]
Thanks @tsuyoshi38 I'll go ahead with some changes on this basis (but I'll add a test for mixed types and non-neutral atom).
